### DISCUSSION TITLE
Dicrete Inversion

### DISCUSSION THREAD

**Discussion by sminize** [gimli-org/pyGIMLi#308] [Ideas]
Hi everyone,
I want to invert my velocity field on only discrete values. Is it possible to constrain the inversion in such a way?

For example, in the Crosshole traveltime tomography example are three velocities present: https://www.pygimli.org/_examples_auto/seismics/plot_3_crosshole_tomography.html

Is it possible to constrain the inversion to these three velocities? Thank you for your kind response.

Best regards

Simon Schmid

**Comment by halbmy** [comment]
Do you mean taking the mesh as it is (knowing the geometry) and invert only for the velocities of the three regions? This would be a nice and simple example...

**Comment by sminize** [comment]
Thank you for your reply. 
Not quite. I was thinking of that the velocities in the object are known, but not where they occur. This would maybe be easier if only two velocities occur in the object (binary reconstruction). In that manner the inversion could be constrained to these two velocities. I am not sure if this makes sense and whether approaches exist for this.

Knowing the geometry and only inverting the velocities also sound interesting.

**Comment by florian-wagner** [comment]
Hi @sminize,

you mean like a multinary inversion (e.g., http://www.cemi.utah.edu/PDF_70_10/2013e.pdf). Yes, that would be great, but is not implemented in pyGIMLi.

Cheers
Florian

**Reply by sminize** [reply]
Hi Florian,
thank you for your reply. 
Thats exactly what i meant. Thank you for the referenced article.

Best regards

Simon
